process/guardian/guardedAccount.go

```diff
@@ -69,17 +75,21 @@ func (agc *guardedAccount) GetActiveGuardian(uah vmcommon.UserAccountHandler) ([
 }
 
 // SetGuardian sets a guardian for an account
-func (agc *guardedAccount) SetGuardian(uah vmcommon.UserAccountHandler, guardianAddress []byte) error {
-	guardian := &guardians.Guardian{
-		Address:         guardianAddress,
-		ActivationEpoch: agc.currentEpoch + agc.guardianActivationEpochs,
-	}
-
+func (agc *guardedAccount) SetGuardian(uah vmcommon.UserAccountHandler, guardianAddress []byte, txGuardianAddress []byte) error {
 	stateUserAccount, ok := uah.(state.UserAccountHandler)
 	if !ok {
 		return process.ErrWrongTypeAssertion
 	}
 
+	if len(txGuardianAddress) > 0 {
+		return agc.instantSetGuardian(stateUserAccount, guardianAddress, txGuardianAddress)
+	}
+
+	guardian := &guardians.Guardian{
+		Address:         guardianAddress,
```

Comment:
is there a check on the newGuardianAddress? at leat for 32 bytes

Reply:
In elrond-vm-common/builtInFunctions/setGuardian.go -> checkSetGuardianArgs
there is already a check for:
isGuardianAddrLenOk := len(vmInput.Arguments[0]) == len(senderAddr)

the only place in production code where SetGuardian is called (and the only place where it should be called) is from `func (sg *setGuardian) ProcessBuiltinFunction` which does this check.

Reply:
ok